### DISCUSSION TITLE
Behavior of "pimcore:deployment:classes-rebuild" was changed in Pimcore v11.3.1 but not mentioned in Upgrade Notes and breaks existing deployments

### DISCUSSION THREAD

**Discussion by akno77** [pimcore/pimcore#17416] [Q&A / Support]
We used to deploy our Docker based Pimcore website via the console command "pimcore:deployment:classes-rebuild -c" to rebuild the Data Object Class files. This worked until the last release when on deploy all the Data Object definitions marked as "skipped" and not as "saved".

I had a look at the core code changes and found out, that the behavior was changed in v11.3.1 vy this commit: https://github.com/pimcore/pimcore/commit/b4591f0936f2e448cdebac5e31dc40c2ab29392a

The Class files are now only created if the new "force" parameter is passed. This breaks the whole existing deployment of all projects that deploy their website via docker and have with every deploy no class files available and have to rebuild them.

Why is this change not mentioned in the Upgrade Notes for v11.3.1? The change is not backward compatible and breaks!

**Comment by dpfaffenbauer** [comment]
you can use `bin/console pimcore:build:classes` during docker build to have the classes available without a database

**Reply by akno77** [reply]
We use the command to create the class files and update the db structure....

**Reply by dpfaffenbauer** [reply]
I know, just wanted to mention that there is a better way now. Your docker image is actually incomplete and static analysis would fail since the Pimcore DataObject Classes are missing.

**Comment by brusch** [comment]
@akno77 seems you are right - if you rely on the fact that all class files are getting created when running `pimcore:deployment:classes-rebuild -c` it could be breaking 🤔 

I see to options to fix that: 
1. we change the `force` parameter to be the default behavior again and create a new parameter, that triggers the new behavior, e.g. `only-changed` parameter 
2. we check whether the class files exist and generate them if they are missing, no matter what the `definitionModificationDate` timestamp says. 

WDYT?

**Reply by akno77** [reply]
Option 2 would be good I think, if modifications to the defintions are also modified/saved in existing class files.

**Reply by brusch** [reply]
@akno77 can you maybe put that into a PR for us? That would be awesome - we'll then review asap.  Thanks in advance!
